src/compiler/abstract_compiler.nit

```diff
@@ -93,6 +96,10 @@ redef class ToolContext
 			exit(1)
 		end
 
+		if opt_debug.value and opt_compile_dir.value == null then
```

Comment:
Ok, then do in at the right place and document it. see the initialisation of the variable `auto_remove` in the method `write_and_make`